### PR TITLE
fix(status_endpoint): add default value for hostname

### PR DESCRIPTION
**Pull request by etolbakov**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
This fix addresses the case when `hostname` function is not available in the system where the project is built. 

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link
close https://github.com/GreptimeTeam/greptimedb/issues/1963

### CODE REVIEW

**Comment by etolbakov** [comment]
Hello @killme2008 @xtang @waynexia @MichaelScofield @sunray-ley

Sorry for the introduced issue that caused unpleasant user experience.
I'll do my best to avoid this type issues in future.

Personally, I like suggestion from @sunray-ley to use a special crate for obtaining the `hostname`.
I'm equally happy to remove the `hostname` information at all, if we decide to follow that route we need adjust the dashboard project accordingly. Happy to do it as well.
Could you please take a look at the pr and provide me feedback?

File: tests-integration/tests/http.rs [hunk 1]
```diff
@@ -59,6 +59,7 @@ macro_rules! http_tests {
                 test_metrics_api,
                 test_scripts_api,
                 test_health_api,
+                test_status_api,
```
Comment (waynexia) [inline tests-integration/tests/http.rs:62]: 👍

File: src/servers/Cargo.toml [hunk 2]
```diff
@@ -41,6 +41,7 @@ derive_builder = "0.12"
 digest = "0.10"
 futures = "0.3"
 hex = { version = "0.4" }
+hostname = "0.3.1"
```
Comment (waynexia) [inline src/servers/Cargo.toml:44]: I've checked its [implementation](https://github.com/svartalf/hostname/blob/master/src/nix.rs#L11-L30). It uses the `gethostname` syscall and has no unwrap 👍

**Review by waynexia — APPROVED** [review]

**Comment by MichaelScofield** [comment]
I prefer remove the "hostname": 1. It can be shown in Dashboard, but Dashboard runs locally, users know their hostnames already. 2. It's kinda "private". "hostname" usually has some info about user's name. 3. Can be used in self-hosted status monitoring about GreptimeDB. In this case, I think it's more useful to just return the ip address of the pod of host.

That said, I'm ok with using the `hostname` crate. Just not use the default "localhost". If cannot retrieve the hostname, simply "unknown" is better. Otherwise you can't distinguish the situation of "really can't retrieve the hostname" from "really running at localhost".

**Comment by waynexia** [comment]
>1. It can be shown in Dashboard, but Dashboard runs locally, users know their hostnames already. 2. It's kinda "private". "hostname" usually has some info about user's name.

This is the hostname of build env. E.g., github `worker` or `minipc0`. It can be used to distinguish where the artifact comes from. 

>If cannot retrieve the hostname, simply "unknown" is better.

Agree 👍

**Comment by Fengys123** [comment]
https://github.com/GreptimeTeam/greptimedb/blob/develop/src/servers/build.rs#L17
It looks like this line also needs to be removed.

**Comment by etolbakov** [comment]
Thanks for the feedback! Will update the PR

**Review by killme2008 — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**